### PR TITLE
Feature/dev env

### PR DESCRIPTION
**Pull request by dankelleher**
Add support for deploying a staging environment, using a different program ID

### CODE REVIEW

File: cli/src/commands/read.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ export default class Read extends Command {
     help: flags.help({char: 'h'}),
   }
 
-  static args = [{name: 'channel'}]
+  static args = [{name: 'channel', required: true}]
```
Comment (rado0x54) [inline cli/src/commands/read.ts:11]: Not summer familiar with `@oclif/command` and could not find that option in any docs, but I assume it's there :)

File: client/src/lib/Channel.ts [hunk 2]
```diff
@@ -41,6 +43,16 @@ export class Channel {
     }
     return encryptCEKForVerificationMethod(this.cek, verificationMethod);
   }
+  
+  async hasMember(did: PublicKey):Promise<boolean> {
```
Comment (rado0x54) [inline client/src/lib/Channel.ts:47]: I assume this is not directly related to the dev env.

File: client/src/lib/constants.ts [hunk 3]
```diff
@@ -2,9 +2,12 @@ import { Commitment, PublicKey } from '@solana/web3.js';
...
 // Should equal the contents of program/program-id.md
-export const PROGRAM_ID: PublicKey = new PublicKey(
-  'boxndjnzQZEWbBku3YipL4pchYRc1zi4nNSrFUwawWo'
-);
+const PROD_PROGRAM_ID = 'boxndjnzQZEWbBku3YipL4pchYRc1zi4nNSrFUwawWo';
+const DEV_PROGRAM_ID = 'dbo6b6DDmWTWpLcZM5SoVEf9X8DSJYPjVfD3Ptz6hod';
+
+export const STAGE = (process.env.REACT_APP_STAGE || process.env.NODE_ENV);
+
+export const PROGRAM_ID: PublicKey = STAGE === 'development' ? new PublicKey(DEV_PROGRAM_ID) : new PublicKey(PROD_PROGRAM_ID);
```
Comment (rado0x54) [inline client/src/lib/constants.ts:10]: Maybe we can generalize this to easily support other configurations in the future? Like an overall envConfig object with keys `deveopment`, `production` and the configuration could have a generator function the publicKey: `envConfig['development'].createPublicKey()`.

File: client/test/e2e.test.ts [hunk 4]
```diff
@@ -104,6 +104,37 @@ describe('E2E', () => {
     })
     expect(channelForBob.address).toEqual(channel.address);
   });
+
+  it('adds a user twice to a group channel throws an error the second time', async () => {
```
Comment (rado0x54) [inline client/test/e2e.test.ts:108]: Ok, I assume that is what the hasMember() fixes.

**Review by rado0x54 — APPROVED** [review]
Some comments. but I'd be fine to merge as is.

File: cli/src/commands/read.ts (hunk 1 above)
Comment (dankelleher) [reply cli/src/commands/read.ts:11]: yeah I tested it- Surprised it's not documented

File: client/src/lib/Channel.ts (hunk 2 above)
Comment (dankelleher) [reply client/src/lib/Channel.ts:47]: not directly. I was debugging adding accounts to the new public channel on the dev env. I noticed if you clear your browser cache then re-add the same DID, it would throw an error, so this function is to help with that.

File: client/src/lib/constants.ts (hunk 3 above)
Comment (dankelleher) [reply client/src/lib/constants.ts:10]: Yeah that sounds good.
